ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/HiveMetaStoreAuthorizer.java

```diff
@@ -312,5 +578,13 @@ private String getCurrentUser() {
   private String getCurrentUser(HiveMetaStoreAuthorizableEvent authorizableEvent) {
     return authorizableEvent.getAuthzContext().getUGI().getShortUserName();
   }
+
+  private UserGroupInformation getUGI() {
+    try {
+      return UserGroupInformation.getCurrentUser();
+    } catch (IOException excp) {
```

Comment:
Why do we swallow the exception?

Reply:
Good catch. I didn't know we were just copy and pasting. Other codes using getCurrentUser were either crashing or rethrow, so I've changed it to throw, and more importantly, return false to the skipAuthorization call.